File: Benchmarks/normal/java/spring/testcode/BenchmarkTest16219.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest16219 {

    @GetMapping("/BenchmarkTest16219")
    public void BenchmarkTest16219(@RequestHeader("Referer") String referer, HttpServletRequest request, HttpServletResponse response) throws Exception {
        response.sendError(500, "Internal error");
    }
}

File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_16219.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request
from starlette.responses import JSONResponse


async def BenchmarkTest16219(request: Request):
    referer_value = request.headers.get('referer', '')
    return JSONResponse({'error': 'An internal error occurred'}, status_code=500)
